File: src/main/java/meowing/zen/feats/general/cleanjoin.java
package meowing.zen.feats.general;

import meowing.zen.utils.EventBus;
import meowing.zen.utils.EventTypes;
import meowing.zen.utils.chatutils;
import meowing.zen.featManager;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class cleanjoin {
    private static final cleanjoin instance = new cleanjoin();
    private cleanjoin() {}
    private static final Pattern guild = Pattern.compile("^Guild > (.+) (joined|left).");
    private static final Pattern friend = Pattern.compile("^Friend > (.+) (joined|left).");
    
    public static void initialize() {
        featManager.register(instance, () -> {
            EventBus.register(EventTypes.GameMessageEvent.class, instance, instance::handleGameMessage);
        });
    }
    
    private void handleGameMessage(EventTypes.GameMessageEvent event) {
        String text = chatutils.removeFormatting(event.getPlainText());
        Matcher m = guild.matcher(text);
        if (m.matches()) {
            chatutils.clientmsg(String.format("§8G %s §b%s", "joined".equals(m.group(2)) ? "§2>>" : "§4<<", m.group(1)));
            event.hide = true;
        }
        m = friend.matcher(text);
        if (m.matches()) {
            chatutils.clientmsg(String.format("§8F %s §b%s", "joined".equals(m.group(2)) ? "§2>>" : "§4<<", m.group(1)));
            event.hide = true;
        }
    }
}

File: src/main/kotlin/meowing/zen/feats/general/cleanjoin.kt
package meowing.zen.feats.general

import meowing.zen.utils.EventBus
import meowing.zen.utils.EventTypes
import meowing.zen.utils.chatutils
import meowing.zen.featManager
import java.util.regex.Pattern

object cleanjoin {
    private val guild = Pattern.compile("^Guild > (.+) (joined|left).")
    private val friend = Pattern.compile("^Friend > (.+) (joined|left).")

    @JvmStatic
    fun initialize() {
        featManager.register(this) {
            EventBus.register(EventTypes.GameMessageEvent::class.java, this, this::onGameMessage)
        }
    }


    private fun onGameMessage(event: EventTypes.GameMessageEvent) {
        val text = chatutils.removeFormatting(event.getPlainText())
        var m = guild.matcher(text)
        if (m.matches()) {
            val action = if ("joined" == m.group(2)) "§2>>" else "§4<<"
            chatutils.clientmsg("§8G $action §b${m.group(1)}", true)
            event.hide()
        }
        m = friend.matcher(text)
        if (m.matches()) {
            val action = if ("joined" == m.group(2)) "§2>>" else "§4<<"
            chatutils.clientmsg("§8F $action §b${m.group(1)}", true)
            event.hide()
        }
    }
}